allennlp/modules/token_embedders/bidirectional_token_embedder.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict
+
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.modules import ScalarMix
+from allennlp.modules.text_field_embedders import TextFieldEmbedder
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.modules.seq2seq_encoders import BidirectionalLanguageModelTransformer
+from allennlp.nn.util import device_mapping, get_text_field_mask, remove_sentence_boundaries
+
+
+@TokenEmbedder.register('bidirectional_token_embedder')
+class BidirectionalTokenEmbedder(TokenEmbedder):
+    """
+    Compute a single layer of representations from a bidirectional language model with a
+    transformer contextualizer.
+
+    Parameters
+    ----------
+    weight_file : ``str``, required
+        An model weights file, e.g. best.th, from a BidirectionalLanguageModel trained using a
+        BidirectionalLanguageModelTransformer as a contextualizer.
+    text_field_embedder : ``TextFieldEmbedder``, required
+        Used to embed the character ids we get in ``forward``.
+    contextualizer : ``BidirectionalLanguageModelTransformer``, required
+        Used to "contextualize" the embeddings.
+    dropout : ``float``, optional.
+        The dropout value to be applied to the representations.
+    remove_bos_eos: ``bool``, optional (default: True)
+        Typically the provided token indexes will be augmented with
+        begin-sentence and end-sentence tokens. If this flag is True
+        the corresponding embeddings will be removed from the return values.
+    requires_grad : ``bool``, optional (default: False)
+        If True, compute gradient of bidirectional language model parameters for fine tuning.
+    """
+    def __init__(self,
+                 weight_file: str,
+                 text_field_embedder: TextFieldEmbedder,
```

Comment:
I am curious about the use of TextFieldEmbedder here (rather than just a `TokenEmbedder`), is this because you might want to concatenate multiple different embeddings of your character ids?

Reply:
Obsolete now that I'm loading the LM directly. (The reason was simply because the `BidirectionalLanguageModel` used one, so I had to match it precisely to load the weights.)